### PR TITLE
Allow parathread manager to set max core price

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Allows the parathread manager to specify the max price for an on-demand core, in units of relay chain currency. The default value is unlimited. This is the price limit, not the price, the actual price depends on usage but that's calculated in the relay chain.

### CODE REVIEW

File: pallets/services-payment/src/lib.rs [hunk 1]
```diff
@@ -249,6 +257,31 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        /// Max core price for parathread in relay chain currency
+        #[pallet::call_index(5)]
+        // TODO: weight
+        #[pallet::weight(T::WeightInfo::set_refund_address())]
```
Comment (girazoki) [inline pallets/services-payment/src/lib.rs:264]: UPdate bench

File: runtime/dancebox/src/xcm_config.rs [hunk 2]
```diff
@@ -570,3 +571,11 @@ impl GetPurchaseCoreCall<RelayChain> for EncodedCallToBuyCore {
         }
     }
 }
+
+pub struct GetMaxCorePriceFromServicesPayment;
+
+impl Convert<ParaId, Option<u128>> for GetMaxCorePriceFromServicesPayment {
+    fn convert(para_id: ParaId) -> Option<u128> {
```
Comment (girazoki) [inline runtime/dancebox/src/xcm_config.rs:578]: Let's use a different trait to convert

**Review by fgamundi — APPROVED** [review]

File: pallets/services-payment/src/lib.rs [hunk 3]
```diff
@@ -249,6 +257,30 @@ pub mod pallet {
...
+
+        /// Max core price for parathread in relay chain currency
+        #[pallet::call_index(5)]
+        #[pallet::weight(T::WeightInfo::set_max_core_price())]
+        pub fn set_max_core_price(
+            origin: OriginFor<T>,
+            para_id: ParaId,
+            max_core_price: Option<u128>,
+        ) -> DispatchResultWithPostInfo {
+            T::SetRefundAddressOrigin::ensure_origin(origin, &para_id)?;
```
Comment (fgamundi) [inline pallets/services-payment/src/lib.rs:269]: Might be good to rename `SetRefundAddressOrigin` to `ManagerOrigin` or similar if it's going to be used in multiple calls

Comment (girazoki) [reply pallets/services-payment/src/lib.rs:269]: Yeah I would do that

**Review by girazoki — COMMENTED** [review]
I would just add an integration test if possible (in the xcm-core-buyer with theh emulator)

**Review by girazoki — APPROVED** [review]

**Pull request merged**